presto-native-execution/presto_cpp/main/PrestoServerOperations.cpp

```diff
@@ -185,22 +189,59 @@ std::string PrestoServerOperations::veloxQueryConfigOperation(
   return unsupportedAction(op);
 }
 
-std::string PrestoServerOperations::debugOperation(
+std::string PrestoServerOperations::taskOperation(
     const ServerOperation& op,
     proxygen::HTTPMessage* message) {
+  if (taskManager_ == nullptr) {
+    return "Task Manager not found";
+  }
+  const auto taskMap = taskManager_->tasks();
   switch (op.action) {
-    case ServerOperation::Action::kTask: {
+    case ServerOperation::Action::kGetDetail: {
       const auto id = message->getQueryParam("id");
-      if (!taskManager_) {
-        return "Task Manager not found";
-      }
-      const auto& map = taskManager_->tasks();
-      const auto& task = map.find(id);
-      if (task == map.end()) {
+      const auto& task = taskMap.find(id);
+      if (task == taskMap.end()) {
         return fmt::format("No task found with id {}", id);
       }
       return task->second->toJsonString();
     }
+    case ServerOperation::Action::kListAll: {
+      uint32_t limit;
+      try {
+        const auto& limitStr = message->getQueryParam("limit");
+        limit = limitStr == proxygen::empty_string
+            ? std::numeric_limits<uint32_t>::max()
+            : stoi(limitStr);
+      } catch (std::exception& ex) {
+        VELOX_FAIL(ex.what());
+      }
+      std::stringstream oss;
+      oss << "[";
+      uint32_t count = 0;
+      for (const auto& task : taskMap) {
+        const auto& veloxTask = task.second->task;
+        if (veloxTask == nullptr) {
```

Comment:
Maybe print out the task id but null for velox task state?
task id, velox state?